modules/web_scraper.py

```diff
@@ -0,0 +1,238 @@
+import bs4
+import requests as req
+import os
+from datetime import date, datetime
+import emoji
+
+_base_URL = "https://mobile.twitter.com/search?q="
+_end_URL_part = "&s=typd&x=0&y=0"
+
+def _get_soup(URL: str):
+    """
+    Downloads the html from given URL and returns the content as soup (bs4)
+    """
+    data = req.get(URL).content
+    soup = bs4.BeautifulSoup(data, "html.parser")
+    return soup
+
+
+def _create_tweet_object(tweet_element, hashtags):
+    # Create a tweet object and store the raw text and the hashtags searched for in it
+    tweet_object = {}
+    tweet_object["search_hashtags"] = hashtags
+    tweet_object["raw_text"] = _extract_inner_text(tweet_element)
+    # Creating tweet_urls property which is a list of strings
+    tweet_urls = _extract_data_urls(tweet_element)
+    tweet_object["tweet_urls"] = tweet_urls
+    # Creating emojis which is a list of strings
+    tweet_emoji_descriptions = _emoji_description_extractor(tweet_object["raw_text"])
+    tweet_object["emojis"] = tweet_emoji_descriptions
+    # Creating the date property
+    tweet_date = _get_tweet_date(tweet_element)
+    tweet_object["date"] = tweet_date
+    return tweet_object
+
+
+def _create_tweet_objects(soup: bs4.BeautifulSoup, hashtags):
+    """
+    Extracts tweets from given soup and returns an array of strings, each string containing a tweet.
+    """
+    tweet_elements = soup.find_all("table", class_=["tweet"])
+    tweets = []
+
+    for element in tweet_elements:
+        tweet_object = _create_tweet_object(element, hashtags)
+        tweets.append(tweet_object)
+    # Return result
+    return tweets
+
+
+def _extract_inner_text(tweet_element):
+    """
+    Extracts raw text from a tweet element.
+    """
+    # The content of the tweet is put into several divs so we find all those divs and extract their inner text
+    tweet_fractions = tweet_element.find_all("div", class_=["dir-ltr"])
+    tweet_text = ""
+    for element in tweet_fractions:
+        tweet_text += str(element.text)
+    # Create initial tweet object and return it
```

Comment:
Return value has changed since previous version but I forgot to change the comment accordingly. Right now it obviously just returns the tweet text as a string and nothing else.